src/lib/recursive_match.js

```diff
@@ -4,33 +4,40 @@
 function find_recursive(opener, closer, text) {
   var out = [];
   var last = [];
-  var chars = text.split('');
+  const chars = text.split('');
   var open = 0;
   for (var i = 0; i < chars.length; i++) {
+    const c = text[i];
     //increment open tag
-    if (chars[i] === opener) {
+    if (c === opener) {
       open += 1;
     }
     //decrement close tag
-    if (chars[i] === closer) {
+    else if (c === closer) {
```

Comment:
I'm assuming here that opener and closer can never be the same, which I think is reasonable.